### DISCUSSION TITLE
Edit tracks and documentation about the mtrack format

### DISCUSSION THREAD

**Discussion by lenzls** [andreynovikov/trekarta#185] [Q&A] [answered]
Hello,

I have quite often the need that I want to edit the finished track. Mainly to remove some captured waypoints, either because of a GPS error or because I forgot to turn off the mapping while I was already driving home.

How could I go about this?

----

I assume this is currently not possible within Trekarta, or even within the scope of the application?

Related to that, is there some documentation on the mtrack data format? I couldn't really find information on it, so I assume it is a self-made format. Is it possible to edit the data outside Trekarta?

Kind regards,

Simon

**Comment by andreynovikov** [comment, marked as answer]
No, this is not possible, nor I can think of any easy and usable way to do this. You always can export track to GPX or KML format and edit it on PC. mtrack is a binary format using Google protobuf (pbf) encoding. The schema can be found here:
https://github.com/andreynovikov/trekarta/blob/master/app/src/main/java/mobi/maptrek/io/Track.proto
Simple raw reader and writer (without auto-generated classes) can be found here:
https://github.com/andreynovikov/trekarta/blob/master/app/src/main/java/mobi/maptrek/io/TrackManager.java
Feel free to ask further questions.
